### DISCUSSION TITLE
Contact Header Parse Exception

### DISCUSSION THREAD

**Discussion by sanshaw1** [resiprocate/resiprocate#302] [Q&A]
Hello,

I am using resip library (v1.7) from a very long time.

Recently i got an parse error issue  in the INVITE message for contact header
Contact
<sip:10.10.114.146:5060>;pub-gruu="sip:310260999999110@ims.cingularme.com;gr=urn:gsma:imei:89994889-944315-0";+g.3gpp.icsi-ref="urn%3Aurn-7%3A3gpp-service.ims.icsi.mmtel";video;audio;text;+g.3gpp.mid-call;+g.3gpp.srvcc-alerting;+g.3gpp.ps2cs-srvcc-orig-pre-alerting;+g.gsma.callcomposer

Error:
BaseException.cxx:17 $ BaseException at ParseBuffer.cxx:79 ParseBuffer.cxx:79, Parse failed skipped over eof in context: Contact
<sip:10.242.114.146:5060>;pub-gruu="sip:310260999999110@ims.cingularme.com;gr=urn:gsma:imei:86704606-944315-0";+g.3gpp.icsi-ref="urn%3Aurn-7%3A3gpp-service.ims.icsi.mmtel";video;audio;text;+g.3gpp.mid-call;+g.3gpp.srvcc-alerting;+g.3gpp.ps2cs-srvcc-orig-pre-alerting;+g.gsma.callcomposer

Can you please let me know whether this contact header is fine or not? Or this issue has been resolved in latest version.

**Comment by ChuLiqiang** [comment]
you can debug the code to find error parameter.

**Reply by sanshaw1** [reply]
I checked the code. Looks to me it expects a @ in the contact uri.
it is not able to find it and it is not escaping characters and treating the full header value as contact uri.
Like this:
Contact: <sip:10.242.114.146:5060%3E%3Bpub-gruu=%22sip%3A310260999999110@ims.cingularme.com;gr=urn:gsma:imei:86704606-944315-0";video;text;+g.3gpp.icsi-ref="urn%3Aurn-7%3A3gpp-service.ims.icsi.mmtel";audio;+g.3gpp.mid-call;+g.3gpp.srvcc-alerting;+g.3gpp.ps2cs-srvcc-orig-pre-alerting;+g.gsma.callcomposer>
To: <sip:21247;phone-context=311.480.eps.ims.cingularme.com@ims.cingularme.com;user=phone>

**Reply by ChuLiqiang** [reply]
![image](https://user-images.githubusercontent.com/2747722/215397170-9a2c51f8-fefc-4e94-b17f-2dc8e7a5351a.png)
you can reference this example. let send message side modify contact. or you modify code to support this contact.

**Comment by sgodin** [comment]
I didn't test this, but it seems like: 
sip:10.242.114.146:5060%3E%3Bpub-gruu=%22sip%3A310260999999110@ims.cingularme.com
is likely being interpreted as the SIP URI portion, ie:
sip:<user>:<password>@<domain>
where:
user =  10.242.114.146
password = 5060%3E%3Bpub-gruu=%22sip%3A310260999999110
domain = ims.cingularme.com

It's not immediately clear to me if we have a bug in our URI parsing or if the Contact is illegal.  If you could get a user into the header I suspect the issue would go away, ie:
sip:<user>@10.242.114.146:5060

**Reply by sanshaw1** [reply]
yes, adding the user , i am not seeing the issue. Tried testing manually by adding @. But this is directly coming from the device to which i have no control.
